From 3af951abfe2abe389d762452c10bb6bc9e6b9faa Mon Sep 17 00:00:00 2001
From: Wujian Sun <wujian.sun_1@nxp.com>
Date: Wed, 1 Mar 2023 11:43:15 +0800
Subject: [PATCH] Build break for ozone wayland platform

egl update khrones headers caused the break.
https://chromium-review.googlesource.com/c/chromium/src/+/3564000

A workaround that revmove gbm EGLNativeWindowType type

| aarch64-poky-linux-ld.lld: error: undefined symbol:
gl::NativeViewGLSurfaceEGL::NativeViewGLSurfaceEGL(gl::GLDisplayEGL*,
void*, std::__1::unique_ptr<gfx::VSyncProvider,
std::__1::default_deletegfx::VSyncProvider >)
| >>> referenced by gl_surface_wayland.cc:35
(./../../ui/ozone/platform/wayland/gpu/gl_surface_wayland.cc:35)
| >>>
thinlto-cache/Thin-7f2605.tmp.o:(ui::GLSurfaceWayland::GLSurfaceWayland(gl::GLDisplayEGL*,
std::__1::unique_ptr<wl_egl_window, ui::EGLWindowDeleter>,
ui::WaylandWindow*))

Upstream-Status: Inappropriate [i.MX-specific]

Signed-off-by: Wujian Sun <wujian.sun_1@nxp.com>
---
 third_party/khronos/EGL/eglplatform.h | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/third_party/khronos/EGL/eglplatform.h b/third_party/khronos/EGL/eglplatform.h
index 99362a23deeac..842c87c4d44e7 100644
--- a/third_party/khronos/EGL/eglplatform.h
+++ b/third_party/khronos/EGL/eglplatform.h
@@ -82,12 +82,6 @@ typedef struct wl_display     *EGLNativeDisplayType;
 typedef struct wl_egl_pixmap  *EGLNativePixmapType;
 typedef struct wl_egl_window  *EGLNativeWindowType;
 
-#elif defined(__GBM__)
-
-typedef struct gbm_device  *EGLNativeDisplayType;
-typedef struct gbm_bo      *EGLNativePixmapType;
-typedef void               *EGLNativeWindowType;
-
 #elif defined(__ANDROID__) || defined(ANDROID)
 
 struct ANativeWindow;
-- 
2.39.2.windows.1

